### PR TITLE
add test for reconciling secret of apicast custom policy

### PR DESCRIPTION
**Pull request by mkudlej**

### CODE REVIEW

**Review by mdujava — CHANGES_REQUESTED** [review]
I did received `ERROR testsuite/tests/apicast/parameters/policies/custom_policy/test_custom_policy_injection.py::test_reconcilation[APIcast operator] - IndexError: list index out of range` on both runs that I did.

File: testsuite/tests/apicast/parameters/policies/custom_policy/conftest.py [hunk 1]
```diff
@@ -31,17 +38,51 @@ def policy_settings():
...
+    secrets.do_action("set", ["data", "secret/" + policy_secret, f"--from-literal=example.lua={example}"])
+    staging_gateway.deployment.wait_for()
+
+    def pod_ready(pod):
+        pod.refresh()
+        status = pod.model.status.conditions
+        return all(st["status"] == "True" for st in status)
+
+    wait_until = backoff.on_predicate(backoff.fibo, max_tries=30)
+    pod = staging_gateway.deployment.get_pods().objects()[0]
```
Comment (mdujava) [inline testsuite/tests/apicast/parameters/policies/custom_policy/conftest.py:78]: Error was here.

**Review by mdujava — APPROVED** [review]

File: testsuite/tests/apicast/parameters/policies/custom_policy/conftest.py [hunk 2]
```diff
@@ -31,17 +39,57 @@ def policy_settings():
...
+            obj_status = pod.model.status.conditions
+            status = status and all(st["status"] == "True" for st in obj_status)
+        return status
+
+    wait_until = backoff.on_predicate(backoff.fibo, max_tries=30)
+    with ExitStack() as stack:
+        staging_gateway.openshift.prepare_context(stack)
+        pods = staging_gateway.deployment.get_pods()
+        wait_until(lambda: pods_ready(pods))
+    assert generation + 1 == staging_gateway.deployment.get_generation()
```
Comment (mdujava) [inline testsuite/tests/apicast/parameters/policies/custom_policy/conftest.py:88]: `test_reconcilation[3scale operator]` errors out here, `assert (12 + 1) == 14`

**Review by mdujava — APPROVED** [review]

**Review by JaurbanRH — APPROVED** [review]

**Review by JaurbanRH — APPROVED** [review]

**Review by mdujava — APPROVED** [review]

File: testsuite/tests/apicast/parameters/policies/custom_policy/conftest.py (hunk 2 above)
Comment (dhlavac) [reply testsuite/tests/apicast/parameters/policies/custom_policy/conftest.py]: @mdujava  is it resolved ?

File: testsuite/tests/apicast/parameters/policies/custom_policy/conftest.py (hunk 1 above)
Comment (dhlavac) [reply testsuite/tests/apicast/parameters/policies/custom_policy/conftest.py]: @mdujava  is it resolved ?

**Pull request merged**